src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { TooltipProps } from 'reactstrap';
+import Tooltip from './Tooltip';
+
+
+interface TruncatedTextProps extends TooltipProps {
+  text: string;
+  targetId: string;
+  recordId: string | number;
+  charLimit: number;
+  tooltip: boolean;
+}
+
+const TruncatedText = ({ text, targetId, recordId, charLimit = 50, tooltip = true, placement = 'top', ...props }: TruncatedTextProps) => {
+  return (
```

Comment:
For others that are not making their own ids https://github.com/appfolio/react-gears/blob/d50ab4cd0a10295ecba6ebf5413e8cf9d4b39f17/src/components/FormChoice.js#L11-L13.
```suggestion
  const [id] = useState(id || getId())

  return (
```